opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/drools/DroolsAlarmContext.java

```diff
@@ -456,10 +456,11 @@ private void eagerlyInitializeAlarm(OnmsAlarm alarm) {
             } catch (ObjectNotFoundException ex) {
                 // This may be triggered if the event attached to the alarm entity is already gone
                 LoggerFactory.getLogger("NMS-15734").warn("Event gone while processing alarm.", new Exception(ex));
-                LoggerFactory.getLogger("NMS-15734").warn("Event: id={}, uei={}, src={}",
+                LoggerFactory.getLogger("NMS-15734").warn("Event: id={}, uei={}, src={}, alarmid={}",
                         alarm.getLastEvent().getId(),
                         alarm.getLastEvent().getEventUei(),
-                        alarm.getLastEvent().getEventSource());
+                        alarm.getLastEvent().getEventSource(),
+                        alarm.getId()==null?"":alarm.getId());
```

Comment:
logger calls `Object.toString(...)` on arguments, which returns `"null"` for `null` objects. Thus making the null-check redundant here.

Reply:
ah that's for pointing that out.